### PR TITLE
Update ASFF parser to create endpoints

### PR DESCRIPTION
**Pull request by vd0n**
### Description
In this update to the ASFF parser, endpoints are now created based solely on the EC2 instance's private IP address. I've implemented a check to ensure each IP address is private, thereby disregarding any public IPs present in the data. This provides a more accurate representation of the internal infrastructure and prevents potential data leaks by not exposing public IPs.

### Test Results
- [x] The updated parser has been tested with ASFF data containing a mix of private and public IP addresses.
- [x] Comprehensive tests have been added to the test suite in `tests/` to cover these changes. (Only check if you've added tests.)

### Documentation
- [ ] I've updated the relevant sections in the [documentation folder](https://github.com/DefectDojo/django-DefectDojo/tree/dev/docs) to reflect these parser changes. (Only check if you've updated the documentation.)

### Checklist
- [x] I have rebased my PR against the latest `dev` branch.
- [x] This feature has been submitted against the `dev` branch.
- [x] The PR has been given a meaningful name.
- [x] My code is flake8 compliant.
- [x] My code is compatible with python 3.11.
- [x] Applicable tests have been added to the unit tests suite.
- [x] I have categorized the PR with the proper label.

### Additional Notes
This modification ensures that DefectDojo users get an accurate representation of their infrastructure without exposing potential security risks associated with public IP addresses.

For moderators:
Please review and let me know if any further changes are needed. Given the sensitive nature of IP addresses, I believe this is a crucial update to the parser.

**Labels:** enhancement, Import Scans

### CODE REVIEW

File: dojo/tools/asff/parser.py [hunk 1]
```diff
@@ -2,19 +2,33 @@
...
-from dojo.models import Finding
+from dojo.models import Endpoint, Finding
 
-# https://docs.aws.amazon.com/securityhub/1.0/APIReference/API_Severity.html
 SEVERITY_MAPPING = {
-    "INFORMATIONAL": "Info",  # No issue was found.
-    "LOW": "Low",  # The issue does not require action on its own.
-    "MEDIUM": "Medium",  # The issue must be addressed but not urgently.
-    "HIGH": "High",  # The issue must be addressed as a priority.
-    # The issue must be remediated immediately to avoid it escalating.
```
Comment (Maffooch) [inline dojo/tools/asff/parser.py:13]: Please don't remove comments that are adding context to the parser

File: dojo/tools/asff/parser.py [hunk 2]
```diff
@@ -29,25 +43,38 @@ def get_description_for_scan_types(self, scan_type):
...
+                test=test
             )
+
+            if "Resources" in item:
+                endpoints = []
+                for resource in item["Resources"]:
+                    if resource["Type"] == "AwsEc2Instance" and "Details" in resource:
+                        details = resource["Details"]["AwsEc2Instance"]
+                        for ip in details.get("IpV4Addresses", []):
+                            if is_private_ipv4(ip):
```
Comment (Maffooch) [inline dojo/tools/asff/parser.py:73]: Why are only private IPs allowed to be added?

**Review by Maffooch — CHANGES_REQUESTED** [review]
Please add some unit tests that successfully import endpoints in the way that you are expecting them to be added

File: dojo/tools/asff/parser.py [hunk 3]
```diff
@@ -29,25 +43,38 @@ def get_description_for_scan_types(self, scan_type):
...
+                unique_id_from_tool=item.get("Id"),
+                test=test
             )
+
+            if "Resources" in item:
+                endpoints = []
+                for resource in item["Resources"]:
+                    if resource["Type"] == "AwsEc2Instance" and "Details" in resource:
+                        details = resource["Details"]["AwsEc2Instance"]
+                        for ip in details.get("IpV4Addresses", []):
```
Comment (kiblik) [inline dojo/tools/asff/parser.py:72]: Can you process IPv6 as well? Not show stopper. Just idea.

**Review by kiblik — CHANGES_REQUESTED** [review]

File: dojo/tools/asff/parser.py [hunk 4]
```diff
@@ -29,25 +43,38 @@ def get_description_for_scan_types(self, scan_type):
...
+                            if is_private_ipv4(ip):
+                                endpoints.append(Endpoint(host=ip))
```
[the remark this answers is not in the document]
Comment (kiblik) [reply dojo/tools/asff/parser.py:86]: If the rest of the community will accept it, okay.

Can you use use different "private address checker"? More standard and easier to maintain.
```suggestion
                            from netaddr import IPAddress
                            if IPAddress(ip).is_private():
                                endpoints.append(Endpoint(host=ip))
```

**Comment by Maffooch** [comment]
There was some hoakey issues with the unit tests for a minute. Please pull the latest dev branch with the updated changes

**Comment by mtesauro** [comment]
@vd0n Can you fix the failing tests  and reviews so this can be merged?

**Comment by manuel-sommer** [comment]
@vd0n will you finish the PR or shaal I create a New one @mtesauro ?

**Comment by mtesauro** [comment]
@manuel-sommer It's been long enough that I'm fine with closing this as stale and having you work on it if you're willing.

**Comment by mtesauro** [comment]
@manuel-sommer I'm going to close this one - feel free to take it up if you have the bandwidth.

**Comment by manuel-sommer** [comment]
Yes, see linked PR above

**Pull request closed without merging**